packages/valory/skills/abstract_round_abci/base.py

```diff
@@ -2051,11 +2051,19 @@ def _schedule_round(self, round_cls: AppState) -> None:
 
         # self.state will point to last result,
         # or if not available to the initial state
-        last_result = (
-            self._round_results[-1]
-            if len(self._round_results) > 0
-            else self._initial_synchronized_data
-        )
+        if self._round_results:
+            synchronized_data_class = getattr(
+                round_cls, "synchronized_data_class", None
+            )
+            if not synchronized_data_class:
+                self.logger.warning(f"No `synchronized_data_class` set on {self}")
+                last_result = self._round_results[-1]
+            else:
+                synchronized_data_class = round_cls.synchronized_data_class  # type: ignore
```

Comment:
```suggestion

```

Don't we already have this from:

https://github.com/valory-xyz/open-autonomy/blob/05fb30f861114c2f1fb9b8731c748356596b7cb1/packages/valory/skills/abstract_round_abci/base.py#L2055-L2057

?

Reply:
yes we do indeed, since i changed from `hasattr` to `getattr` this line of code is excessive

Reply:
remove this unnecessary code. this is messy enough as it stands

Reply:
The issue is clearly quite involved. Why not simply apply the fix in the contribution service, like I did in [termination_abci](https://github.com/valory-xyz/open-autonomy/blob/main/packages/valory/skills/termination_abci/behaviours.py#L124), and leave this for a different time. I am afraid that no fix will be acceptable without introducing breaking changes.